addons/project_mrp/models/mrp_production.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+
+
+class MrpProduction(models.Model):
+    _inherit = 'mrp.production'
+
+    task_id = fields.Many2one('project.task', string='Task', readonly=True)
+
+    def _compute_analytic_distribution(self):
+        super()._compute_analytic_distribution()
+        analytic_account = self.env['account.analytic.account']
+        if self._context.get('task_id'):
```

Comment:
Small nitpick, but it's slightly more correct to use `self.env.context` rather than `self._context`, although it doesn't matter much, so up to you